internal/mode/static/state/conditions/conditions.go

```diff
@@ -583,3 +588,23 @@ func NewNginxGatewayInvalid(msg string) conditions.Condition {
 		Message: msg,
 	}
 }
+
+// NewNginxProxyValid returns a Condition that indicates that the NginxProxy config is valid.
+func NewNginxProxyValid() conditions.Condition {
+	return conditions.Condition{
+		Type:    string(ngfAPI.NginxProxyConditionValid),
+		Status:  metav1.ConditionTrue,
+		Reason:  string(ngfAPI.NginxProxyReasonValid),
+		Message: "NginxProxy is valid",
+	}
+}
+
+// NewNginxProxyInvalid returns a Condition that indicates that the NginxProxy config is invalid.
+func NewNginxProxyInvalid(msg string) conditions.Condition {
```

Comment:
what if we have two conditions - one is Valid (or perhaps accepted, to follow Gateway API) and another one is Programmed. This will reflect that NginxProxy affects data plane configuration - it can be valid from validation perspective, but the data plane might fail to reload.